tests/tfhe_rust/iterators.mlir

```diff
@@ -0,0 +1,110 @@
+// RUN: heir-translate %s --emit-tfhe-rust | FileCheck %s
+
+// CHECK: enum GateInput
+
+// CHECK: enum OpType
+
+module {
+  // CHECK-LABEL: pub fn parallel_loop
+  // CHECK-NEXT:   [[sks:v[0-9]+]]: &ServerKey,
+  // CHECK-NEXT:   [[input:v[0-9]+]]: &[i16; 2],
+  // CHECK-NEXT: ) -> {{[[][[]}}i8; 8]; 2] {
+  // CHECK-NEXT: let mut temp_nodes : HashMap<usize, Ciphertext> = HashMap::new();
+  // CHECK-NEXT: let mut luts : HashMap<&str, LookupTableOwned> = HashMap::new();
+  // CHECK:      let mut run_level
+
+  // CHECK:      let mut [[v1:.*]] : HashMap<(usize, usize), i8> = HashMap::new();
+  // CHECK-NEXT: for [[v2:.*]] in 0..2 {
+  // CHECK-NEXT:   let [[v3:.*]] = [[input]][[[v2]]];
+  // CHECK-NEXT:   for [[v4:.*]] in 0..8 {
+  // CHECK-NEXT:     let [[v5:.*]] = [[v4]] as i16;
+  // CHECK-NEXT:     let [[v6:.*]] = [[v5]] & [[v3]];
+  // CHECK-NEXT:     let [[v7:.*]] = [[v6]] as i8;
+  // CHECK-NEXT:     [[v1]].insert(([[v2]] as usize, [[v4]] as usize), [[v7]]);
+  // CHECK-NEXT:     }
+  // CHECK-NEXT:  }
+  // CHECK-NEXT: core::array::from_fn
+  // CHECK-SAME:   [[v1]].get(&([[i0:.*]], [[i1:.*]])).unwrap().clone()
+  func.func @parallel_loop(%sks : !tfhe_rust.server_key, %arg0 : memref<2xi16>) -> (memref<2x8xi8>) {
```

Comment:
mildly confused because nothing in this test seems parallel? It looks like it should be called something like `generate_cleartext_ops`

Reply:
ah, this was named back when this was emitted as a parallel loop.